### DISCUSSION TITLE
Custom Module Minimal API with AD Authentication

### DISCUSSION THREAD

**Discussion by Sharafudeen** [oqtane/oqtane.framework#3247] [General] [answered]
Hi,

The purpose of having a custom module is to integrate (consume APIs and expose minimal APIs) with other systems and it has separate windows AD authentication. Does Oqtane can allow to extend or I can use existing OpenID Connect provider and configure windows AD?

**Comment by sbwalker** [comment]
Oqtane supports OIDC - it can be configured for each site in an installation.

![image](https://github.com/oqtane/oqtane.framework/assets/4840590/e211acb6-7f3d-43fe-b5e1-dcad0c3b892c)

**Comment by Sharafudeen** [comment]
@sbwalker To consume third party system’s APIs we need to authenticate Windows AD with service account which I needs to configure “AD Path” and service account credentials (instead of Client ID & Secret). How do I configure this?

**Reply by sbwalker** [reply]
What you are describing does not sound like a standard OIDC configuration. If the third party system does not support OIDC or OAuth2 then you will need to build your own custom integration.

**Reply by Sharafudeen** [reply]
I don’t want to change or extend Oqtane core authentication provider instead can I build custom authentication & authorization in Oqtane’s module level?

**Reply by sbwalker** [reply, marked as answer]
Yes, it should be possible to add the Windows AD integration to a custom module. You may need to include some custom middleware as well which can be integrated using IServerStartup.

**Reply by Sharafudeen** [reply]
Wow.. amazing..!!

This means Oqtane has a capabilities to extend third party systems integration without affecting Oqtane’s core!

Thanks a lot..!!
